### PR TITLE
Add configuration for directory environments

### PR DESCRIPTION
**Pull request by xaque208**
This work allows the the user to take advantage of directory environments by providing new params on the server config section and providing the logic necessary to remove old puppet.conf configuration items and replacing them with the new style configuration options. Old config styles have been deprecated and will be removed in future versions of Puppet.

### CODE REVIEW

**Comment by igalic** [comment]
looks (still) good to me.
anyone else wanna take a look before i hit that merge button?

File: manifests/server/config.pp [hunk 1]
```diff
@@ -9,21 +9,42 @@
     section => 'master',
   }
 
-  if $puppet::server::environmentpath {
-    $env_ensure = 'present'
-    $mod_ensure = 'absent'
+  if $puppet::server::directoryenvs == true or $puppet::server::directoryenvs == 'true' {
```
Comment (tampakrap) [inline manifests/server/config.pp]: Instead of this, how about validate_bool in server.pp, and here check only `if $puppet::server::directoryenvs`?

**Comment by tampakrap** [comment]
How about also removing the deprecated `modulepath`, `manifest` and `config_version`?

File: manifests/server/config.pp (hunk 1 above)
Comment (igalic) [inline manifests/server/config.pp]: :+1:

**Comment by xaque208** [comment]
I say lets remove old config options as part of another PR.  We've got piles of cleanup needed on this module imo.

**Pull request merged**